platform/web/retry/tests/middleware_test.go

```diff
@@ -0,0 +1,140 @@
+package tests
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/fewlinesco/go-pkg/platform/web/retry"
+)
+
+func TestRetryRoundTripperMiddleware(t *testing.T) {
+	type roundTripperMiddlewareTestCase struct {
+		name              string
+		expectedHTTPCode  int
+		httpCodesToReturn []int
+		expectedCalls     int
+		config            retry.Config
+	}
+
+	delay := time.Second / 2 // half a second
```

Comment:
Considering how quick the requests should be without any kind of I/O happening, we can probably lower the delay at least to 250ms without issues. That would make the tests faster and no less correct imho

Reply:
fair enough 👍